frontend/components/MainContent/MainContent.tsx

```diff
@@ -30,32 +34,66 @@ const MainContent = ({
   const {
     sandboxExpiry,
     config,
-    isSandboxMode,
     isPremiumTier,
     noSandboxHosts,
+    apnsExpiry,
+    abmExpiry,
   } = useContext(AppContext);
 
-  const isAppleBmTermsExpired = config?.mdm?.apple_bm_terms_expired;
-
   const sandboxExpiryTime =
     sandboxExpiry === undefined
       ? "..."
       : formatDistanceToNow(new Date(sandboxExpiry));
 
-  const showAppleABMBanner =
-    isAppleBmTermsExpired && isPremiumTier && !isSandboxMode;
+  const renderAppWideBanner = () => {
+    const isAppleBmTermsExpired = config?.mdm?.apple_bm_terms_expired;
+    const isApplePnsExpired = hasLicenseExpired(apnsExpiry || "");
+    const willApplePnsExpireIn30Days = willExpireWithinXDays(
+      apnsExpiry || "",
+      30
+    );
+    const isAppleBmExpired = hasLicenseExpired(abmExpiry || "");
+    const willAppleBmExpireIn30Days = willExpireWithinXDays(
+      abmExpiry || "",
+      30
+    );
+    const isFleetLicenseExpired = hasLicenseExpired(
+      config?.license.expiration || ""
+    );
+
+    if (isPremiumTier) {
+      if (isApplePnsExpired || willApplePnsExpireIn30Days) {
+        return <ApplePNCertRenewalMessage expired={isApplePnsExpired} />;
+      }
+
+      if (isAppleBmExpired || willAppleBmExpireIn30Days) {
+        return <AppleBMRenewalMessage expired={isAppleBmExpired} />;
+      }
+
+      if (isAppleBmTermsExpired) {
+        return <AppleBMTermsMessage />;
+      }
+
+      if (isFleetLicenseExpired) {
+        return <LicenseExpirationBanner />;
+      }
+    }
 
+    return <></>;
+  };
   return (
     <div className={classes}>
-      {showAppleABMBanner && <AppleBMTermsMessage />}
-      <SandboxGate
-        fallbackComponent={() => (
-          <SandboxExpiryMessage
-            expiry={sandboxExpiryTime}
-            noSandboxHosts={noSandboxHosts}
-          />
-        )}
-      />
+      <div className={`${baseClass}--animation-disabled`}>
+        {renderAppWideBanner()}
+        <SandboxGate
```

Comment:
we can take out the `SandboxGate`

Reply:
Limiting scope as I'm leaving in 5 work hours.